src/test/java/org/llorllale/cactoos/matchers/HasPropertyTest.java

```diff
@@ -0,0 +1,145 @@
+import org.hamcrest.core.AllOf;
+import org.hamcrest.core.IsNot;
+import org.hamcrest.text.IsEqualIgnoringCase;
+import org.hamcrest.text.StringContainsInOrder;
+import org.junit.Test;
+
+/**
+ * Test case for {@link HasProperty}.
+ *
+ * @since 1.0.0
+ * @checkstyle ClassDataAbstractionCoupling (100 lines)
+ */
+public final class HasPropertyTest {
+
+    /**
+     * Simple positive case.
+     */
+    @Test
+    public void positiveMatch() {
+        new Assertion<>(
+            "must match 'a=1'",
+            new ScalarHasValue<>(new HasProperty("a", "1")),
+            new Matches<>(new PropertiesOf("a=1"))
+        ).affirm();
+    }
+
+    /**
+     * Simple positive case only for key.
+     */
+    @Test
+    public void positiveMachKey() {
+        new Assertion<>(
+            "must match 'b=...'",
+            new ScalarHasValue<>(new HasProperty("b")),
+            new Matches<>(new PropertiesOf("b=..."))
+        ).affirm();
+    }
+
+    /**
+     * Positive case with Matchers.
+     */
+    @Test
+    public void positiveFuzzyMatch() {
+        new Assertion<>(
+            "must match 'f=q'",
+            new ScalarHasValue<>(
+                new HasProperty(
+                    new IsEqualIgnoringCase("F"),
+                    new IsEqualIgnoringCase("Q")
+                )
+            ),
+            new AllOf<>(
+                new ListOf<Matcher<? super ScalarHasValue<Properties>>>(
+                    new Matches<>(new PropertiesOf("F=q")),
+                    new Matches<>(new PropertiesOf("f=Q")),
+                    new Matches<>(new PropertiesOf("f=q"))
+                )
+            )
+        ).affirm();
+    }
+
+    /**
+     * Negative case.
+     */
+    @Test
+    public void negativeMatch() {
+        new Assertion<>(
+            "must mismatches 'abc=2' & 'xyz=1'",
+            new ScalarHasValue<>(new HasProperty("abc", "1")),
+            new AllOf<>(
+                new ListOf<Matcher<? super ScalarHasValue<Properties>>>(
+                    new IsNot<>(new Matches<>(new PropertiesOf("abc=2"))),
+                    new IsNot<>(new Matches<>(new PropertiesOf("xyz=1")))
+                )
+            )
+        ).affirm();
+    }
+
+    /**
+     * Test for mismatch description readability.
+     *
+     * @todo #67:30m Make {@link Matcher#describeMismatch} of the original
+     *  matcher is called by {@link ScalarHasValue}.
+     *  And then adjust this test with more readable message originating
+     *  from {@link HasProperty}.
+     */
+    @Test
+    public void describesCorrectly() {
+        new Assertion<>(
+            "must throw an exception that describes the properties",
+            () -> {
+                new Assertion<>(
+                    "",
+                    new PropertiesOf("b=2"),
+                    new ScalarHasValue<>(new HasProperty("c", "3"))
+                ).affirm();
+                return true;
+            },
+            new Throws<>(
```

Comment:
@andreoss please use `Mismatches` instead of testing throws

Reply:
Done